src/routing/Routing_Py_Adapter.cpp

```diff
@@ -29,48 +29,37 @@ Routing_Py_Adapter::Routing_Py_Adapter(std::string t_route_connection_path, std:
  * Parameterized constructor Routing_Py_Adapter without the flow_vector
  *
  * @param t_route_connection_path
- * @param input_path
- * @param catchment_subset_ids
+ * @param t_route_config_file_with_path
  * @param number_of_timesteps
  * @param delta_time
  */
-Routing_Py_Adapter::Routing_Py_Adapter(std::string t_route_connection_path, std::string input_path,
-                                       const std::vector<std::string> &catchment_subset_ids,
+Routing_Py_Adapter::Routing_Py_Adapter(std::string t_route_connection_path,
+                                       std::string t_route_config_file_with_path,
                                        int number_of_timesteps, int delta_time)
 {
 
-  //Cast vector of catchment_subset_ids to Python list 
-  py::list catchment_subset_ids_list = py::cast(catchment_subset_ids);
-
   //Append the t_route_connection_path
   utils::ngenPy::InterpreterUtil::addToPyPath(t_route_connection_path);
 
-  //Append the input_path
-  utils::ngenPy::InterpreterUtil::addToPyPath(input_path);
-
-  //Leave this call here for now for calling previous version of module
-  //this->t_route_module = py::module_::import("next_gen_network_module");
+  //Import ngen_main
+  this->t_route_module = py::module_::import("ngen_main");
 
-  //Import next_gen_route_main
-  this->t_route_module = py::module_::import("next_gen_route_main");
+  std::vector<std::string> arg_vector;
 
-  //Create object for the set_paths subroutine
-  py::object set_paths = t_route_module.attr("set_paths");
+  arg_vector.push_back("-f");
 
-  //Call set_paths subroutine
-  set_paths(t_route_connection_path);
+  arg_vector.push_back(t_route_config_file_with_path);
 
-  //Create object for ngen_routing subroutine
-  py::object ngen_routing = t_route_module.attr("ngen_routing");
+  //Cast vector of args to Python list 
+  py::list arg_list = py::cast(arg_vector);
 
-  //Call ngen_routing subroutine
-  ngen_routing(number_of_timesteps, delta_time);
+  //Create object for the ngen_main subroutine
+  py::object ngen_main = t_route_module.attr("ngen_main");
 
-  //Create object for call_read_catchment_lateral_flows subroutine
-  py::object call_read_catchment_lateral_flows = t_route_module.attr("call_read_catchment_lateral_flows");
+  //Call ngen_main subroutine
+  ngen_main(arg_list);
```

Comment:
Working through this, the syntax is a little different from what I've seen a lot of, but I think that is ok.

But, I'm not sure it will work.  That module (and function) imports something out of `nwm_routing`, but I don't think there is anything that ensures that will be in the interpreter's system path.

This one might be complicated enough that we need to talk through it to make sure we are on the same page.